src/main/java/com/iota/iri/model/safe/ByteSafe.java

```diff
@@ -1,5 +1,10 @@
 package com.iota.iri.model.safe;
 
+/**
+ * Extends the HashSafeObject class to generate a null safe byte array with retrievable hash code index. This uses
+ * Double-Checked Locking to ensure that the byte object is initialised correctly when referenced, and does not
```

Comment:
Double-Checked Locking is only mentioned here, without any reference to where this happens.
 Maybe better to move this statement to the SafeObject class and mention that it should use Double-Checked Locking somewhere (SafeObject doesnt do this, or my definition is off)

Reply:
@galrogo I'm having a hard time interpreting the Double-Check locking with respect to its implementation here. I'm not sure I understand where the DCL is happening here. The `ByteSafe` object is constructed using the `HashSafeObject` as an indexing means for the `SafeObject`.  Is the Double-Checking occurring as a result of the nesting of the `SafeObject` within the `HashSafeObject` within the `ByteSafe`?

Reply:
https://en.wikipedia.org/wiki/Double-checked_locking

If you want more explanation let's have a call :-)